tripy/nvtripy/frontend/ops/slice.py

```diff
@@ -105,23 +117,43 @@ def __getitem__(
     if isinstance(index, Tensor):
         return gather(self, 0, index)
 
-    index = make_list(index)
+    index = [None] if index is None else make_list(index)
     ellipsis_count = index.count(Ellipsis)
+    new_dim_count = index.count(None)
     if ellipsis_count > 1:
         raise_error("Slicing index can only have a single ellipsis ('...')")
-    if len(index) > self.rank:
+    if len(index) > self.rank + new_dim_count:
         raise_error(f"Input tensor has a rank of {self.rank} but was attempted to be sliced with {len(index)} indices")
     if ellipsis_count:
         ellipsis_idx = index.index(Ellipsis)
-        num_slices = self.rank - len(index) + 1
+        num_slices = self.rank + new_dim_count - len(index) + 1
         index[ellipsis_idx : ellipsis_idx + 1] = [slice(None)] * num_slices
 
     inp_shape = self.shape
+    if new_dim_count:
+        new_shape = []
+        slice_indices = []
+        current_dim = 0
+        for idx in index:
+            if idx is None:
+                new_shape.append(1)
+                slice_indices.append(slice(None))
```

Comment:
Since we know the shape is `1` it might actually reduce the IR further to use `slice(0, 1)` instead of `slice(None)` which needs to infer bounds.